### PR TITLE
Javascript-Capstone

### PR DESCRIPTION
**Pull request by vanelnw**
In this project, we :

- create an E-commerce application an implement likes and comment functionality .
- Use JavaScript to manipulate DOM elements.
- Use JavaScript events.
- Use callbacks and promises.
- Use webpack to bundle JavaScript.\
- Use npm as a package manager.
- Use API documentation.
- Send and receive data from an API.
- Write unit tests for a JavaScript app.
- Use the AAA pattern for unit tests.

### CODE REVIEW

File: src/style.css [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+.hidden {
+  display: none;
+}
+
+.card-image {
+  max-width: 100%;
+}
+
+.commentSection > * {
```
Comment (rloterh) [inline src/style.css:170]: - [ ] OPTIONAL
Your comment overlay appears to be very opaque. Overlays generally have some form transparent background, and likewise, I think you apply some amount of transparent effect to the background of your comment overlay. 

**Your comment overlay:**
image

File: src/modules/commentsCounter.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const countElement = () => {
```
Comment (rloterh) [inline src/modules/commentsCounter.js:1]: - [ ] Your comment counter is not working as desired. When a comment is added the counter is not updated immediately, until the page is refreshed. Kindly ensure that the comment counter updates the same time as the comment entered by a user is persisted.
- [ ] Also, as per the design guidelines, comments should appear above the form.

**Comments counter delay:**
image

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+const products = [];
+
+const ProductList = document.querySelector('.app_content');
+
+const store = async () => {
+  const response = await fetch(
+    'https://fakestoreapi.com/products/category/electronics',
+  ).then((res) => res.json());
+
+  const Likes = await fetch(
```
Comment (rloterh) [inline src/index.js:16]: - [ ]  OPTIONAL
I think you can improve upon your _likes_ features to make it appear more realistic in these ways:
    - Like icon should toggle to indicate _like_ or _unlike_ when a particular user clicks on the like icon. If the page is refreshed, then assume that new user is present.
    - Total count of likes per user/site visitor should be 1. So, let's assume a user clicks a like, and the number of likes for a particular item should be `+1`, and if the same user clicks the like again the number of likes for that same object should be `-1`. If the page is refreshed, then assume that new user is present. 😉

File: README.md [hunk 4]
```diff
@@ -1 +1,212 @@
...
+- **Add new comment to the favorite products**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+
+- [Live Demo Link](https://vanelnw.github.io/JavaScript-Capstone/)
```
Comment (rloterh) [inline README.md:114]: - [ ] OPTIONAL
Live demo is pointing to the ReadMe. Kindly ensure to update your demo link accordingly.

File: README.md [hunk 5]
```diff
@@ -1 +1,212 @@
...
+
+    `npm start`
+
+### Run tests
+
+To run tests, run the following command:
+
+    `npm test`
+
+
```
Comment (rloterh) [inline README.md:168]: - [ ] OPTIONAL
 Kindly ensure to add the _Future Features_ to your ReadMe, which should include 1-3 potential features to be added

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Vanel & Raymond, 

You did extremely great on this project! 👏 👏
There are some issues that you still need to work on to prepare your project for the assessment but you are almost there 🚦 🚦

## Suggested changes
_Kindly check my inline comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._ 😃 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 6]
```diff
@@ -1 +1,224 @@
...
+- **Add new reservation to the favorite products**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+
+- [Live Demo Link](https://vanelnw.github.io/JavaScript-Capstone/)
```
Comment (OyePriscilla) [inline README.md:114]: - [ ] [optional] Kindly update your` live demo` to display your app properly. Perhaps you may want to build your app with `gh-pages.` You can read through [this short tutorial](https://github.com/turingschool-examples/webpack-starter-kit/blob/main/gh-pages-procedure.md).

**Review by OyePriscilla — APPROVED** [review]
Hi @vanelnw  and @dialite 😃, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

GloriousTenderHalcyon-max-1mb

## Highlights
Descriptive pull request message and title :heavy_check_mark: 
Linter error checks validated :heavy_check_mark: 
Commit messages are meaningful. :heavy_check_mark: 
GitHub flow observed ✔️

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by dialite** [comment]
### Hi @vanelnw 

In my opinion, the  project approval is a well deserved one. Congratulations.! 🎉🎉 🎉

### Suggestion

Please in reaction to the optional change that was suggested by the reviewer, I will suggest if we can add the '/dist' to the end of the link so that GitHub page deploy can locate the 'index.html' file which is required. 

Thank you.
Once again Congratulations! 🎉 🎉 🎉 🎉 🎉 🎉 🎉

**Pull request merged**